service.go

```diff
@@ -423,16 +442,62 @@ func populateService(k8sService *k8score.Service, acks map[string]string) servic
 			log.WithError(err).Warnf("Cannot parse isDaemon label value for service with name %s.", serviceName)
 		}
 	}
+	appPort := getAppPortForService(k8sService)
+	hs.populateSystemCodeForService(serviceName, appPort)
 
 	return service{
 		name:        serviceName,
-		appPort:     getAppPortForService(k8sService),
+		appPort:     appPort,
 		isDaemon:    isDaemon,
 		isResilient: isResilient,
 		ack:         acks[serviceName],
 	}
 }
 
+func (hs *k8sHealthcheckService) populateSystemCodeForService(serviceName string, servicePort int32) {
+	go func(serviceName string, servicePort int32, services *servicesMap, guard chan struct{}) {
+		guard <- struct{}{}
+		defer func(g chan struct{}) {
+			<-g
+		}(guard)
+		log.Infof("Fetching system code for service %s", serviceName)
+
+		req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s:%d/__health", serviceName, servicePort), nil)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to prepare system code request for service %s", serviceName)
+			return
+		}
+		resp, err := hs.httpClient.Do(req)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to execute system code request for service %s", serviceName)
+			return
+		}
+		if resp.StatusCode != http.StatusOK {
+			log.WithError(err).Errorf("System code request returned non-200 code [%d] for service %s", resp.StatusCode, serviceName)
+			return
+		}
+		defer func() {
+			err := resp.Body.Close()
+			if err != nil {
+				log.WithError(err).Errorf("Failed to close close response body reader for service %s.", serviceName)
+			}
+		}()
```

Comment:
**suggestion:** We can move this `defer` before `if resp.StatusCode != http.StatusOK`, because even if the response is non-200, we still want to close the body.